main_test.go

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package section is responsible for splitting a program into its different
+// sections without attempting to parse the contents.s
+package main
+
+import (
+	"bytes"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/pkg/diff"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Arguments struct {
+	args []string
+	fset *token.FileSet
+}
+
+var command = Arguments{
+	args: []string{"-l", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"},
+	fset: token.NewFileSet(),
+}
+
+func TestNewArgParser(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify correct initialization through newArgParser", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		outputStringLint := argParser.FindOptionByLongName("lint").Description
+		outputStringLintShort := argParser.FindOptionByShortName(rune('l')).Description
+		expectedStringLint := "Turn on lint flag to output patch matches to stdout instead of modifying the files"
+		assert.Equal(t, expectedStringLint, outputStringLint)
+		assert.Equal(t, expectedStringLint, outputStringLintShort)
+		assert.Equal(t, false, opts.Linting)
+		assert.Equal(t, []string(nil), opts.Patches)
+
+	})
+
+	t.Run("Verify functionality of newArgParser", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		patchFileName := command.args[2]
+		assert.Equal(t, true, opts.Linting)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+		require.NoError(t, err)
+
+	})
+
+}
+
+func TestLoadPatches(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify correct functionality of LoadPatches", func(t *testing.T) {
```

Comment:
Applies to all tests:
subtest naming: This is overly verbose. "verify", "correct", and "functionality" are all implied from the fact that we're writing a test.

We can just call this "success" if we're verifying that loadPatches does not fail (since we're inside TestLoadPatches). Maybe we can also test that the returned `patches` has a specific length?